src/auth/routes.rs

```diff
@@ -1,88 +1,36 @@
-use actix_web::*;
-use diesel;
-use diesel::prelude::*;
-use diesel::r2d2::{ConnectionManager, Pool};
-use uuid;
+use actix::prelude::*;
+use actix_web::{
+    http, middleware, server, App, AsyncResponder, FutureResponse, HttpResponse, Path, Error, HttpRequest,
+    State, HttpMessage, error, Json
+};
+use crate::AppState;
 
-use actix_web::actix::*;
 
-use actix_web::error::*;
-use actix_web::error;
+use futures::{future, Future, Stream};
 
-use crate::models;
-use crate::state::AppState;
-use crate::config::SECRET;
-use crate::oauth;
-
-pub struct GetToken {
-    pub grant_type: String,
-    pub username: String,
-    pub password: String,
-
-}
-
-impl Message for GetToken {
-    type Result = Result<oauth::Token, Error>;
-}
-
-pub struct Authorise {
-    pub username: String,
-    pub password: String,
-}
-
-impl Message for Authorise {
-    type Result = Result<oauth::Token, Error>;
+#[derive(Debug, Serialize, Deserialize, Message)]
+pub struct GetTokenReq {
+    username: String,
+    grant_type: String,
+    password: String,
 }
 
+pub fn get_token((item, state): (Json<GetTokenReq>, State<AppState>)) -> impl Future<Item = HttpResponse, Error = Error> {
 
-impl Handler<GetToken> for super::DbExecutor {
-    type Result = Result<oauth::Token, Error>;
-
-    fn handle(&mut self, msg: GetToken, ctx: &mut Self::Context) -> Self::Result {
-        use crate::schema::users::dsl::*;
-
-        let conn: &SqliteConnection = &self.0.get().unwrap();
-
-        let mut items = users
-            .filter(name.eq(&msg.username))
-            .load::<models::user::User>(conn)
-            .map_err(|_| error::ErrorInternalServerError("Error loading person"))?;
-
-        //TODO: fix unwrap
-        let user = items.pop().unwrap();
-
-
-        let claims = oauth::Claims{
-            username: msg.username.clone(),
-        };
-
-        let access_token = oauth::generate_token(claims, SECRET.to_string());
-
-        Ok(oauth::Token {
-            access_token: access_token.to_string(),
-            token_type: oauth::TokenType::Bearer,
-            expires_in: 1500,
+    let copy = item.into_inner();
 
+    state
+        .db
+        .send(super::db::GetToken {
+            grant_type: "password".to_string(),
+            username: copy.username.clone(),
+            password: copy.password.clone(),
         })
-    }
-}
-//
-//impl Handler<VerifyToken> for DbExecutor {
-//    type Result = Result<oauth::Token, Error>;
-//
-//    fn handle(&mut self, msg: GetToken, _: &mut Self::Context) -> Self::Result {
-//        use self::schema::users::dsl::*;
-//
-//
-//    }
-//}
+        .from_err()
+        .and_then(|res| match res {
+            Ok(user) => Ok(HttpResponse::Ok().json(user)),
+            Err(_) => Ok(HttpResponse::InternalServerError().into()),
```

Comment:
RFC6749 define specific error codes, here is some inspiration from what I did 5 years ago: https://github.com/ng-bottle-libraries/oauth2-rfc6749/blob/master/rfc6749/oauth2_errors.py